src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/r/SignatureToolTipManager.java

```diff
@@ -81,9 +145,45 @@ public void initialize(UIPrefs uiPrefs,
       server_ = server;
    }
    
+   private void attachPreviewHandler()
+   {
+      detachPreviewHandler();
+      preview_ = Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent preview)
+         {
+            if (preview.getTypeInt() == Event.ONMOUSEMOVE)
+            {
+               NativeEvent event = preview.getNativeEvent();
+               coordinates_ = new MouseCoordinates(
+                     event.getClientX(),
+                     event.getClientY());
+            }
+            else if (preview.getTypeInt() == Event.ONKEYDOWN)
+            {
+               coordinates_ = null;
+            }
+            
+            timer_.schedule(DELAY_MS);
```

Comment:
This makes sense, but do we really need to worry about the cost of scheduling a timer? (I imagine that's a fairly cheap operation).

@jmcphers is definitely right pointing out that we don't want to re-schedule on _all_ events, though.